Modules/System/Cryptography Management/src/Rfc2898DeriveBytes.Codeunit.al

```diff
@@ -0,0 +1,31 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+codeunit 50101 "Rfc2898DeriveBytes"
+{
+    procedure FuncCreateHashWithRfc2898DeriveBytes(Password: Text; InputText: Text; PsuedoRandomNumber: Integer; FillCharacter: Integer) Hash: Text
+    var
```

Comment:
Move the implementation code into https://github.com/microsoft/ALAppExtensions/blob/master/Modules/System/Cryptography%20Management/src/CryptographyManagementImpl.Codeunit.al

This codeunit acts as a facade layer in between, calling that new function.

Reply:
done!

Reply:
Facade layers do not contain implementations, as they're the APIs being called from other extensions.
While the Impl layer, "Cryptography Management Impl." contains the implementation of said APIs.

This function should only contain one variable for the impl codeunit (CryptographyManagementImpl) and one line CryptographyManagementImpl.HashRfc2898DeriveBytes(….).
While the CryptographyManagementImpl.HashRfc2898DeriveBytes function contains all the logic we have here right now.